### ISSUE TITLE
Getting error when no label is present in validation data 

### ISSUE THREAD

**Issue by nishcreed** [ultralytics/ultralytics#9445]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

I finetuned the model with my custom dataset and everything worked fine. Later, i realized that there is no image with no object in the training or validation dataset (which I need to cover all types of data). So, i separately formed training and validation datasets with no object . I trained the previously finetuned model for 5 more epochs with the new object-less dataset. I get  the following error:
![yolo error](https://github.com/ultralytics/ultralytics/assets/88657287/093e9165-3a13-430b-b997-d5c83c46b921)

### Environment

Google colab

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@nishcreed validation requires labels, suggest you merge your two datasets into one and train on that.

**Comment by nishcreed** [comment]
> @nishcreed validation requires labels, suggest you merge your two datasets into one and train on that.

Fine

**Comment by nishcreed** [comment]
Fine

**Comment by glenn-jocher** [comment]
Hi @nishcreed! 👋

Thanks for reaching out. To tackle the issue with images having no objects, you indeed need labeled data for both scenarios (images with and without objects) merged into one dataset for training and validation. YOLO expects at least some annotations, even if it's just an empty file to signify no objects in certain images.

You can merge your current datasets and include empty label files for those images without objects. Here's a quick tip on making an empty label file for an image without objects:
```bash
touch path/to/your/dataset/labels/img_without_objects.txt
```
Remember, the name of the empty label file should match the image file.

Hope this helps! Feel free to follow up if you have more questions. Happy training! 😊

**Comment by nishcreed** [comment]
> Hi @nishcreed! 👋
> 
> Thanks for reaching out. To tackle the issue with images having no objects, you indeed need labeled data for both scenarios (images with and without objects) merged into one dataset for training and validation. YOLO expects at least some annotations, even if it's just an empty file to signify no objects in certain images.
> 
> You can merge your current datasets and include empty label files for those images without objects. Here's a quick tip on making an empty label file for an image without objects:
> 
> ```shell
> touch path/to/your/dataset/labels/img_without_objects.txt
> ```
> 
> Remember, the name of the empty label file should match the image file.
> 
> Hope this helps! Feel free to follow up if you have more questions. Happy training! 😊

Ok thanks

**Comment by glenn-jocher** [comment]
You're welcome, @nishcreed! If you have any more questions or run into issues, don't hesitate to reach out. Happy training! 😊
